### PR TITLE
Add boundary test coverage for resync query pagination

### PR DESCRIPTION
**Pull request by bbrks**
Adds extra coverage for the boundary conditions around resync query pagination and assert on resync processed stat.

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `DCP resync,^TestResyncAllTombstones$,GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1952/
- [x] `DCP resync,^TestResyncAllTombstones$,GSI=false,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1953/
- `^TestResyncAllTombstones$,GSI=false,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1954/
  - View engine `unknown_error` ... rerun:
  - [x] https://jenkins.sgwdev.com/job/SyncGateway-Integration/1956/
- [ ] `^TestResyncAllTombstones$,GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1955/

### CODE REVIEW

File: rest/changestest/changes_api_test.go [hunk 1]
```diff
@@ -3866,41 +3865,46 @@ func TestResyncAllTombstones(t *testing.T) {
...
+			rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+				DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+					QueryPaginationLimit: base.IntPtr(queryPaginationLimit),
+					Unsupported:          &db.UnsupportedOptions{UseQueryBasedResyncManager: true}}},
+			})
+			rt.GetDatabase().PurgeInterval = 0
+			defer rt.Close()
 
-	TestResyncTombstones := func(numDocs int) {
+			for i := 0; i < numTombstones; i++ {
```
Comment (mohammed-madi) [inline rest/changestest/changes_api_test.go:3889]: I believe all test cases are testing the case where querylimit = numberoftombstones, can we add an extra doc every time queryPaginationLimit is even to test querylimit < numberoftombstones

Comment (bbrks) [reply rest/changestest/changes_api_test.go:3889]: `queryPaginationLimit` is fixed to 5 in the const at the top of the test, and `numTombstones` varies around that in either direction.

Comment (mohammed-madi) [reply rest/changestest/changes_api_test.go:3889]: nevermind

**Review by mohammed-madi — APPROVED** [review]
